### ISSUE TITLE
Make RawRepresentable types automatically conform to QueryParameter

### ISSUE THREAD

**Issue by florianreinhart** [novi/mysql-swift#75]
It would be nice to use `enum`s, `OptionSet`s and other `RawRepresentable` types directly as query parameters.

This could make use of conditional conformance introduced in Swift 4.1. The following code will not compile and complain that an extension for a protocol cannot have an inheritance clause: `Extension of protocol 'RawRepresentable' cannot have an inheritance clause`. However, it should explain my idea. I guess mysql-swift needs some special internal handling of `RawRepresentable`s to make this work.

```swift
extension RawRepresentable: QueryParameter where RawValue: QueryParameterType {
    public func queryParameter(option: QueryParameterOption) -> QueryParameterType {
        return self.rawValue
    }
}
```

**Comment by novi** [comment]
I have implemented experimental codes into mysql-swift.  (88d2f5c5a8e9604fdc4d5998610e0f94f03d74c6)
But it will case error `multiple matching functions named...` when  structs have both `Codable` and `QueryParameter` protocols. And it will be avoided by this fix 9514552af251f3265f82f1fd3fbc6fb40744a1a8.

**Comment by florianreinhart** [comment]
Thanks @novi! LGTM, though I haven't tested the new code 😉

**Comment by novi** [comment]
Released 0.9.0 and will close the ticket.
